sidecar/pkg/sidecar/sidecar_handlers.go

```diff
@@ -0,0 +1,56 @@
+// This file is part of MinIO Operator
+// Copyright (c) 2024 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package sidecar
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/minio/operator/pkg/common"
+)
+
+func configureSidecarServer(c *Controller) *http.Server {
+	router := mux.NewRouter().SkipClean(true).UseEncodedPath()
+
+	router.Methods(http.MethodPost).
+		Path(common.SidecarAPIConfigEndpoint).
+		HandlerFunc(c.CheckConfigHandler).
+		Queries(restQueries("c")...)
+
+	router.NotFoundHandler = http.NotFoundHandler()
+
+	s := &http.Server{
+		Addr:           "0.0.0.0:" + common.SidecarHTTPPort,
```

Comment:
Any specific reason using `0.0.0.0` and not `127.0.0.1` as in sidecar/pkg/sidecar/bucket_dns_handlers.go

Reply:
the next change that will come is to have the operator check on the sidecar about if a configuration change has been consolidated before issuing a `mc admin service restart`, ths service will be protected by mTLS and an API Key, so `0.0.0.0` is fine